src/cunumeric/utilities/dispatch.h

```diff
@@ -0,0 +1,192 @@
+          std::forward<Fnargs>(args)...);
+      }
+      case CuNumericTypeCodes::CUNUMERIC_TYPE_POINT5: {
+        return f.template operator()<CuNumericTypeCodes::CUNUMERIC_TYPE_POINT5, DIM>(
+          std::forward<Fnargs>(args)...);
+      }
+      case CuNumericTypeCodes::CUNUMERIC_TYPE_POINT6: {
+        return f.template operator()<CuNumericTypeCodes::CUNUMERIC_TYPE_POINT6, DIM>(
+          std::forward<Fnargs>(args)...);
+      }
+      case CuNumericTypeCodes::CUNUMERIC_TYPE_POINT7: {
+        return f.template operator()<CuNumericTypeCodes::CUNUMERIC_TYPE_POINT7, DIM>(
+          std::forward<Fnargs>(args)...);
+      }
+      case CuNumericTypeCodes::CUNUMERIC_TYPE_POINT8: {
+        return f.template operator()<CuNumericTypeCodes::CUNUMERIC_TYPE_POINT8, DIM>(
+          std::forward<Fnargs>(args)...);
+      }
+      case CuNumericTypeCodes::CUNUMERIC_TYPE_POINT9: {
+        return f.template operator()<CuNumericTypeCodes::CUNUMERIC_TYPE_POINT9, DIM>(
+          std::forward<Fnargs>(args)...);
+      }
+      default: {
+        return legate::inner_type_dispatch_fn<DIM>{}(code, f, std::forward<Fnargs>(args)...);
+      }
+    }
+    assert(false);
+    return f.template operator()<CuNumericTypeCodes::CUNUMERIC_TYPE_POINT1, DIM>(
+      std::forward<Fnargs>(args)...);
+  }
+};
+
+template <typename TypeCode, typename Functor, typename... Fnargs>
+constexpr decltype(auto) double_dispatch(int dim, TypeCode code, Functor f, Fnargs&&... args)
+{
+  switch (dim) {
+#if LEGION_MAX_DIM >= 1
+    case 1: {
+      return cunumeric::inner_type_dispatch_fn<1>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 2
+    case 2: {
+      return inner_type_dispatch_fn<2>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 3
+    case 3: {
+      return inner_type_dispatch_fn<3>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 4
+    case 4: {
+      return inner_type_dispatch_fn<4>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 5
+    case 5: {
+      return inner_type_dispatch_fn<5>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 6
+    case 6: {
+      return inner_type_dispatch_fn<6>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 7
+    case 7: {
+      return inner_type_dispatch_fn<7>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 8
+    case 8: {
+      return inner_type_dispatch_fn<8>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+#if LEGION_MAX_DIM >= 9
+    case 9: {
+      return inner_type_dispatch_fn<9>{}(code, f, std::forward<Fnargs>(args)...);
+    }
+#endif
+  }
+  assert(false);
+  return inner_type_dispatch_fn<1>{}(code, f, std::forward<Fnargs>(args)...);
+}
+
+template <typename Functor, typename... Fnargs>
+constexpr decltype(auto) double_dispatch(int dim1, int dim2, Functor f, Fnargs&&... args)
```

Comment:
where are you using this?

Reply:
I am not using this, but might in the future. Would you recommend removing currently unused functions?

Reply:
what use case do you have in mind?